sink.go

```diff
@@ -145,6 +147,10 @@ func (sr *sinkRegistry) newFileSinkFromURL(u *url.URL) (Sink, error) {
 		return nil, fmt.Errorf("file URLs must leave host empty or use localhost: got %v", u)
 	}
 
+	if strings.Contains(u.Path, "..") {
```

Comment:
A filename can contain `..`, so I don't think this check is right
```
$ vim foo..bar

$ cat foo..bar
hello
```

should we instead look for an absolute path before calling `newFileSinkFromPath`?

Reply:
newFileSinkFromPath must support relative paths. Paths without the file:// form are allowed to be relative. This is documented and tested already. 

Paths that take the file:// form will use net/url, and the path will always be absolute (because u.Path will always start with / for them).

The ask was to specifically reject path traversals with the URLs. We could check for /../ instead.

TBH, I'm not convinced that there's a vulnerability here, but I'll believe that a security expert could use some combination of symlinks and .. to get arbitrary file access.

Reply:
The path traversal security issue may be a false positive, since we have the hostname check above(`file://foo/bar` parses to host=`foo`, path=`/bar`).

Stepping back, what is the security vulnerability exactly? If the user has control over the paths, they can specify relative paths like `../foo` which is a valid supported path -- does it matter if it's via URL or a relative path?

Reply:
As I said, I'm not convinced that there's a vulnerability here.
But you're right, given that the user has full control of this input, this is even less a question of sanitization.
If we want to appease the linter, we can add a `filepath.Clean` there, but the actual check may be unnecessary.

Reply:
Commented in https://github.com/uber-go/zap/issues/1390#issuecomment-1867932885